lib/kredis/types/hash.rb

```diff
@@ -18,7 +19,9 @@ def []=(key, value)
   end
 
   def update(**entries)
-    hset entries.transform_values { |val| type_to_string(val, typed) }.compact if entries.flatten.any?
+    with_expiration do
```

Comment:
Refreshing expiration on modification differs from mainline Redis behavior.

If we want this, it should probably be distinct from setting an expiry on the key.